### PR TITLE
#167190538 Authenticated User Can Comment on an Existing Article

### PR DESCRIPTION
**Pull request by ayodejiAA**
#### What does this PR do?
- Allows registered users to comment on an existing article
- Allows fetching (Authentication is optional) of all comments on an existing article.

#### Description of Task proposed in this pull request?
- Add route to add comment  -  **POST**  ``` /articles/<slug>/comments``` [Protected Route]
- Add route to get all comments for an article -  **GET**  ``` /articles/<slug>/comments```
- Add comment validation
- Document feature on swagger
- Add migration table for comment

#### How should this be manually tested (Quality Assurance)?
- Checkout to this branch ```ft-user-comment-on-article-167190538```.
- Register or login to get an authentication token. 
- User the add comment route to add a comment. Required: ```comment: "Insert comment"```
- Use the get all comment route to fetch all article comment

#### What are the relevant pivotal tracker stories?
- [#167190538](https://www.pivotaltracker.com/story/show/167190538)

#### Any background context you want to add (Operations Impact)?
Comments allow authors to get feedback on their articles. We all love to know what people think about what we put out to the world.

#### What I have learned working on this feature:
- I learnt Sequelize association

#### Screenshots: 
- Add comment
Screenshot 2019-08-15 at 12 58 29 PM

 - Get all comments
Screenshot 2019-08-15 at 1 00 49 PM

 - Article not found
Screenshot 2019-08-15 at 1 01 06 PM

- Article has no comment
Screenshot 2019-08-15 at 1 01 32 PM

- Mock
Screenshot 2019-08-15 at 3 21 17 PM

### CODE REVIEW

File: server/controllers/Comments.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  /**
+   * @name getAllCommentsForArticle
+   * @async
+   * @static
+   * @memberof Comments
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of new follower
+   */
+  static async getAllCommentsForArticle(req, res) {
```
Comment (chialuka) [inline server/controllers/Comments.js]: Why aren't you using a `try-catch` in this controller?

File: server/routes/comment.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+const route = express.Router();
+
+const { verifyToken, validateCommentBody, getSessionFromToken } = middlewares;
+
+route.post(
+  '/:slug/comments',
+  verifyToken,
+  getSessionFromToken,
+  validateCommentBody,
```
Comment (chialuka) [inline server/routes/comment.js:19]: Add `checkUserVerification` middleware to this route

File: test/comments/comment.test.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+      const res = await chai
+        .request(app)
+        .post(`${baseUrl}/articles/${artilceSlug}/comments`)
+        .set('authorization', commenter.token)
+        .send({
+          comment: new Array(4999).join('a')
+        });
+      expect(res).to.have.status(201);
+      expect(res.body).to.haveOwnProperty('comment');
+      expect(res.body.comment.user.following).to.be.false; // Not following the author
```
Comment (chialuka) [inline test/comments/comment.test.js]: Please remove commented code

Comment (ayodejiAA) [reply test/comments/comment.test.js]: That is not commented code.

File: server/routes/comment.js (hunk 2 above)
Comment (ayodejiAA) [reply server/routes/comment.js:19]: Added

File: server/routes/index.js [hunk 4]
```diff
@@ -15,5 +16,6 @@ route.use('/auth', auth);
 route.use('/profiles', profile, follower);
 route.use('/user', userFollower);
 route.use('/articles', article);
+route.use('/articles', comment);
```
Comment (JuwonAbiola) [inline server/routes/index.js:21]: You can add the comment beside the article instead of repeating route.use ('/articles)

**Review by Adekoreday — DISMISSED** [review]

File: server/controllers/Comments.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      user.following = isUserFollowingAuthor;
+      commentDetail.user = user;
+      return serverResponse(res, 201, { comment: commentDetail });
+    } catch (error) {
+      serverError(res);
+    }
+  }
+
+  /**
+   * @name getAllCommentsForArticle
```
Comment (dinobi) [inline server/controllers/Comments.js]: change to `getArticleComments`

Comment (ayodejiAA) [reply server/controllers/Comments.js]: Done.

**Review by JuwonAbiola — APPROVED** [review]

**Review by henryade — APPROVED** [review]

**Review by Adekoreday — APPROVED** [review]

**Pull request merged**